### PR TITLE
fix #106146 Crash when pressing LEFT in note entry mode

### PR DESCRIPTION
**Pull request by heuchi**
https://musescore.org/en/node/106146

### CODE REVIEW

File: libmscore/input.cpp [hunk 1]
```diff
@@ -116,7 +116,9 @@ void InputState::moveInputPos(Element* e)
             return;
 
       Segment* s;
-      if (e->isChordRest())
+      if (e->isSegment())
```
Comment (lasconic) [inline libmscore/input.cpp]: The else is not useful anymore then.
Also, it's probably a good case to use the new toSegment(), toChord() everywhere in the function https://github.com/musescore/MuseScore/blob/master/mscore3.txt#L70

Comment (heuchi) [inline libmscore/input.cpp]: Updated to follow new guide lines and removed the else branch.
I added a Q_ASSERT to make sure s has been assigned a valid pointer.

**Comment by heuchi** [comment]
The main reason for the crash is the Q_ASSERT statement in Element.isChordRest() in file libmscore/element.h:616
<pre>bool isChordRest() const { Q_ASSERT(type() != Element::Type::SEGMENT);
      return type() == Element::Type::REST || type() == Element::Type::CHORD</pre>

I don't quite understand why we disallow a call to isChordRest() for Segment pointers.

**Comment by wschweer** [comment]
There was aname conflict for isChordRest(); in old code it was used to check Segments (and the segment type). Current code use isChordRest() to test the element type. To find the old uses i added an ASSERT. For legal uses of isChordRest() on segments you have to use isChordRest1(). This is a hack and isChordRest1() will be replaced by isChordRest() and the assert removed.

**Pull request merged**